src/shared/components/mutationTable/column/ascnCopyNumber/ASCNCopyNumberColumnFormatter.tsx

```diff
@@ -0,0 +1,195 @@
+            ascnCopyNumberValue !== ASCNCopyNumberValue.NA &&
+            hasASCNProperty(mutation, 'totalCopyNumber') &&
+            getWGD(sampleIdToClinicalDataMap, mutation.sampleId) !==
+                ASCNCopyNumberValue.NA &&
+            getASCNCopyNumberColor(ascnCopyNumberValue.toString()) !==
+                ASCN_BLACK
+        ) {
+            sampleToCNA[
+                mutation.sampleId
+            ] = mutation.alleleSpecificCopyNumber.totalCopyNumber.toString();
+        } else {
+            sampleToCNA[mutation.sampleId] = ASCNCopyNumberValue.NA;
+        }
+    }
+    return sampleToCNA;
+}
+
+function getSortValue(
+    data: Mutation[],
+    sampleIdToClinicalDataMap: { [key: string]: ClinicalData[] } | undefined,
+    sampleIds: string[]
+) {
+    const displayValuesBySample: {
+        [key: string]: string;
+    } = getAllTotalCopyNumberForMutation(
+        data,
+        sampleIdToClinicalDataMap,
+        sampleIds
+    );
+    const sampleIdsWithValues = sampleIds.filter(
+        sampleId => displayValuesBySample[sampleId]
+    );
+    const displayValuesAsString = sampleIdsWithValues.map(
+        (sampleId: string) => {
+            return displayValuesBySample[sampleId];
+        }
+    );
+    return displayValuesAsString.join(';');
+}
+
+export function getWGD(
+    sampleIdToClinicalDataMap:
+        | { [sampleId: string]: ClinicalData[] }
+        | undefined,
+    sampleId: string
+) {
+    let wgdData = sampleIdToClinicalDataMap
+        ? sampleIdToClinicalDataMap[sampleId].filter(
+              (cd: ClinicalData) => cd.clinicalAttributeId === 'FACETS_WGD'
+          )
+        : undefined;
+    return wgdData !== undefined && wgdData.length > 0
+        ? wgdData[0].value
+        : ASCNCopyNumberValue.NA;
+}
+
+export const getDefaultASCNCopyNumberColumnDefinition = (
+    sampleIds?: string[],
+    sampleIdToClinicalDataMap?: { [sampleId: string]: ClinicalData[] },
+    sampleManager?: SampleManager
+) => {
+    return {
+        name: 'Integer Copy #',
+        render: (d: Mutation[]) =>
+            ASCNCopyNumberColumnFormatter.renderFunction(
+                d,
+                sampleIds ? sampleIds : d.length > 0 ? [d[0].sampleId] : [],
+                sampleIdToClinicalDataMap,
+                sampleManager
+            ),
+        sortBy: (d: Mutation[]) =>
+            getSortValue(
+                d,
+                sampleIdToClinicalDataMap,
+                sampleIds ? sampleIds : d.length > 0 ? [d[0].sampleId] : []
+            ),
+    };
+};
+
+export default class ASCNCopyNumberColumnFormatter {
+    /* Determines the display value by using the impact field.
+     *
+     * @param data  column formatter data
+     * @returns {string}"Clonal" text value
+     */
+    public static renderFunction(
+        data: Mutation[],
+        sampleIds: string[],
+        sampleIdToClinicalDataMap?: { [sampleId: string]: ClinicalData[] },
+        sampleManager?: SampleManager
```

Comment:
also, should this be optional?

Reply:
this is optional because we don't use the sampleManager in the results view. It's only being used to generate the numbered sample icons in the tooltips. (cases where multiple samples under a patient  have same variants)

Reply:
@sheridancbio double check?

Reply:
I remember that there two versions of some of these formatter classes, and the code was refactored so that a single version of the formatter could be used under both scenarios (patient view where the patient has multiple samples versus a sample view where only one sample's mutations are shown). So that is the essential reason why the sample manager may or may not be available.